vars/mavenExecute.groovy

```diff
@@ -0,0 +1,86 @@
+import com.sap.piper.ConfigurationLoader
+import com.sap.piper.ConfigurationMerger
+
+def call(Map parameters = [:]) {
+
+    handlePipelineStepErrors(stepName: 'mavenExecute', stepParameters: parameters) {
+        final script = parameters.script
+
+        prepareDefaultValues script: script
+        final Map stepDefaults = ConfigurationLoader.defaultStepConfiguration(script, 'mavenExecute')
+
+        final Map stepConfiguration = ConfigurationLoader.stepConfiguration(script, 'mavenExecute')
+
+        List parameterKeys = [
+            'dockerImage',
+            'globalSettingsFile',
+            'projectSettingsFile',
+            'pomPath',
+            'flags',
+            'goals',
+            'm2Path',
+            'defines'
+        ]
+        List stepConfigurationKeys = [
+            'dockerImage',
+            'globalSettingsFile',
+            'projectSettingsFile',
+            'pomPath',
+            'm2Path'
+        ]
+
+        Map configuration = ConfigurationMerger.merge(parameters, parameterKeys, stepConfiguration, stepConfigurationKeys, stepDefaults)
+
+        String command = "mvn"
+
+        def globalSettingsFile = configuration.globalSettingsFile
+        if (globalSettingsFile?.trim()) {
+            if(globalSettingsFile.trim().startsWith("http")){
+                downloadSettingsFromUrl(globalSettingsFile)
+                globalSettingsFile = "settings.xml"
+            }
+            command += " --global-settings ${globalSettingsFile}"
```

Comment:
Applies for all file pathes used in the command string: Does this also work when the file path contains blanks? Maybe we should quote the pathes.

Reply:
It is now escaped